app/src/main/java/com/spoony/spoony/core/util/extension/ModifierExt.kt

```diff
@@ -0,0 +1,17 @@
+package com.spoony.spoony.core.util.extension
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+
+@Composable
+inline fun Modifier.noRippleClickable(crossinline onClick: () -> Unit): Modifier = composed {
+    this.clickable(
+        indication = null,
+        interactionSource = remember { MutableInteractionSource() },
+        onClick = { onClick() }
```

Comment:
p3) 이 부분 이렇게도 되지 않나요? 불필요한 람다를 추가로 만들필요는 없을 것 같습니다.

```suggestion
        onClick = onClick
```

Reply:
## 람다 전달과 호출의 차이

람다 전달:
```kotlin
onClick = onClick
```
여기서 onClick은 람다를 전달만 하고 호출은 나중에 이뤄집니다.
clickable 내부에서 이 람다를 호출할 때까지 실행되지 않습니다.

람다 호출:
```kotlin
onClick = { onClick() }
```

전달받은 onClick 람다를 호출하도록 명시적으로 정의합니다.
{} 내부에서 실행 로직을 추가하거나 호출을 제어할 수 있습니다.

### 왜 inline과 crossinline을 사용했는가?
inline: 함수 호출 오버헤드를 줄이고, noRippleClickable 함수가 자주 호출될 때 성능을 최적화.
전달받은 onClick 람다에 대한 익명 객체 생성을 방지.
crossinline: onClick에서 비지역 반환이 발생할 가능성을 방지.
예를 들어, return을 사용해 noRippleClickable 바깥의 함수를 종료하려고 할 때 오류를 방지.

### 결론
전달받은 람다를 명시적으로 호출해야 하기 때문에 onClick 을 사용하지 않고 { onClick() } 로 해야된다고 합니다.

<img width="741" alt="image" src="https://github.com/user-attachments/assets/d4f4f745-e011-4709-8b1c-50e76a142f37" />

일단 잘몰라서 정리해봤는데.. 아직도 사실 이해가 잘 가진 않습니다.

Reply:
이부분은 저도 잘 모르던 부분인데 찾아보고 공유하도록 하겠습니다!